File: 프로그래머스/2/118667. 두 큐 합 같게 만들기/두 큐 합 같게 만들기.py
from collections import deque
def solution(queue1, queue2):
    q1 = deque(queue1)
    q2 = deque(queue2)
    
    sum1 = sum(q1)
    sum2 = sum(q2)
    total = sum1 + sum2
    
    if total % 2 != 0:
        return -1
    
    target = total // 2
    cnt = 0
    max_cnt = len(queue1) * 3
    
    while cnt <= max_cnt:
        if sum1 == target:
            return cnt
        if sum1 > target:
            moveNum = q1.popleft()
            sum1 -= moveNum
            q2.append(moveNum)
            sum2 += moveNum
        else:
            moveNum = q2.popleft()
            sum2 -= moveNum
            q1.append(moveNum)
            sum1 += moveNum
        cnt += 1
    return -1

File: 프로그래머스/2/118667. 두 큐 합 같게 만들기/두 큐 합 같게 만들기.js
function solution(queue1, queue2) {
    const arr = [...queue1, ...queue2];
    const n = queue1.length;
    
    const q1 = [...queue1];
    const q2 = [...queue2];
    
    let sum1 = q1.reduce((acc, cur) => acc + cur, 0);
    let sum2 = q2.reduce((acc, cur) => acc + cur, 0);
    const total = sum1 + sum2;
    
    if(total % 2 !== 0) return -1;
    const target = total / 2;
    
    let p1 = 0;
    let p2 = n;
    
    let cnt = 0;
    const maxCnt = queue1.length * 3;
    
    while(cnt <= maxCnt){
        if(sum1 === target) return cnt;
        
        if(sum1 > target){
            sum1 -= arr[p1];
            sum2 += arr[p1];
            p1++;
        } else {
            sum2 -= arr[p2];
            sum1 += arr[p2];
            p2++;
        }
        cnt++;
    }
    
    return -1;
}